src/WingetCreateTests/WingetCreateTests/UnitTests/UpdateCommandTests.cs

```diff
@@ -185,8 +185,53 @@ public async Task UpdateFailsWithUnmatchedPackages()
             var updatedManifests = await command.DeserializeExistingManifestsAndUpdate(initialManifestContent);
             Assert.IsNull(updatedManifests, "Command should have failed");
             string result = this.sw.ToString();
-            Assert.That(result, Does.Contain(Resources.MultipleInstallerUpdateDiscrepancy_Error), "Installer discrepency error should be thrown");
-            Assert.That(result, Does.Contain(string.Format(Resources.MissingPackageError_Message, InstallerType.Msix, InstallerArchitecture.X86)), "Missing package error should be thrown");
+            Assert.That(result, Does.Contain(Resources.NewInstallerUrlMustMatchExisting_Message), "Installer must have match error should be thrown");
+        }
+
+        /// <summary>
+        /// Since some installers are incorrectly labeled on the manifest, resort to using the installer URL to find matches.
+        /// This unit test uses a msi installer that is not an arm64 installer, but because the installer URL includes "arm64", it should find a match.
+        /// </summary>
+        /// <returns>A <see cref="Task"/> representing the asynchronous unit test.</returns>
+        [Test]
+        public async Task UpdateBasedOnInstallerUrlMatch()
+        {
+            TestUtils.InitializeMockDownloads(
+                "arm64/" + TestConstants.TestMsiInstaller,
+                "arm/" + TestConstants.TestMsiInstaller,
+                "win64/" + TestConstants.TestMsiInstaller,
+                "win32/" + TestConstants.TestMsiInstaller);
+            (UpdateCommand command, var initialManifestContent) = GetUpdateCommandAndManifestData("TestPublisher.MatchWithInstallerUrl", null, this.tempPath, null);
+            var initialManifests = Serialization.DeserializeManifestContents(initialManifestContent);
+            var initialInstaller = initialManifests.SingletonManifest.Installers.First();
+
+            var updatedManifests = await command.DeserializeExistingManifestsAndUpdate(initialManifestContent);
+            Assert.IsNotNull(updatedManifests, "Command should have succeeded");
+            foreach (var updatedInstaller in updatedManifests.InstallerManifest.Installers)
+            {
+                Assert.AreNotEqual(initialInstaller.InstallerSha256, updatedInstaller.InstallerSha256, "InstallerSha256 should be updated");

```

Comment:
Instead of just checking !=, let's check if each one matches what is expected, that way we can make sure the URL parsing logic works. You can do this by making the archs an array:

var archs = new[] { "arm64", "arm"... };
TestUtils.InitializeMockDownloads(archs.Select(a => $"{a}/{TestConstants.TestMsiInstaller}));

and then down below you can assert that each expected arch is present in the Installers

Reply:
Thanks for the suggestion. That is a much more exhaustive test. I did a little modification so now the test makes sure the architecture is parsed correctly.